lib/AST/Expr.cpp

```diff
@@ -1243,7 +1251,12 @@ CaptureListExpr *CaptureListExpr::create(ASTContext &ctx,
                                          ClosureExpr *closureBody) {
   auto size = totalSizeToAlloc<CaptureListEntry>(captureList.size());
   auto mem = ctx.Allocate(size, alignof(CaptureListExpr));
-  return ::new(mem) CaptureListExpr(captureList, closureBody);
+  auto *expr = ::new(mem) CaptureListExpr(captureList, closureBody);
+
+  for (auto capture : captureList)
+    capture.Var->setParentCaptureList(expr);
+
+  return expr;
```

Comment:
I would prefer putting the for loop *before* the `CaptureListExpr` constructor to compute the nullable `CaptureListExpr`, then passing it in to the constructor. That way the instance variable could be a `const` and it would be a bit clearer when reading the code that it is immutable.

Reply:
I'm not sure I understand. The for loop references the newly-constructed CaptureListExpr instance.

Reply:
Sorry, my mistake. Thanks for pointing it out so gently.

Reply:
I guess the real issue is not knowing where the var is captured when the var is created. Would be great if the var could be immutable somehow.